app/components/energy_tariff_table_component.rb

```diff
@@ -5,17 +5,44 @@ class EnergyTariffTableComponent < ViewComponent::Base
 
   delegate :can?, :cannot?, to: :helpers
 
-  def initialize(id: 'tariff-table', tariff_holder:, tariffs:, show_actions: true)
+  def initialize(id: 'tariff-table', tariff_holder:, tariffs:, show_actions: true, show_prices: true)
     @id = id
     @tariff_holder = tariff_holder
     @tariffs = tariffs
     @show_actions = show_actions
+    @show_prices = show_prices
+  end
+
+  def show_prices?
+    @show_prices
   end
 
   def show_meters?
     @tariff_holder.school?
   end
 
+  def can_toggle_status?(energy_tariff)
+    @tariff_holder.site_settings? || energy_tariff.dcc?
+  end
+
+  def can_delete?(energy_tariff)
+    return false if energy_tariff.dcc?
+    return false if @tariff_holder.site_settings? && energy_tariff.enabled?
+    true
+  end
+
+  def show_actions?
+    @show_actions
+  end
+
+  def start_time(price)
+    price.start_time.to_s(:time)
+  end
+
+  def end_time(price)
+    price.start_time.to_s(:time)
```

Comment:
should this be `price.end_time.to_s(:time)` (end time instead of start time)

Reply:
d'oh